extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/SplitPackageProcessor.java

```diff
@@ -94,7 +94,8 @@ void splitPackageDetection(ApplicationArchivesBuildItem archivesBuildItem,
                             }
                         }
                     } else {
-                        splitPackages.add(a.getGroupId() + ":" + a.getArtifactId());
+                        splitPackages.add(a.getGroupId() + ":" + a.getArtifactId() + ":" + a.getType() +
+                                (a.getClassifier().isEmpty() ? "" : (":" + a.getClassifier())));
```

Comment:
I think it's possible that `a.getClassifier()` returns null, or?

Reply:
I don't think so, see https://github.com/quarkusio/quarkus/blob/main/independent-projects/bootstrap/app-model/src/main/java/io/quarkus/bootstrap/model/AppArtifactKey.java#L99

Reply:
actually, you could have simply called `a.toString()` :)

Reply:
Heh, didn't realize that :D
Fixed.